File: c/demos/bubbleSort.c
#include <stdio.h>
#include "../utils/printArray.h"
#include "../sorters/bubbleSort.h"

int main () {
    int a[] = { 9, 234, 12, 333, 21, 8 };
    const int n = 6;
    printf("Array's length is %d\n", n);
    printf("Before sorting:\n");
    printArray(a, n);
    bubbleSort(a, n);
    printf("After sorting:\n");
    printArray(a, n);
    return 0;
}

File: javascript/bubbleSort.js
function bubbleSort (a, n) {
    for (let i = n -1; i > 0; i --) {
        for (let j = 0; j< n - 1; j ++) {
            if (a[j] > a[j+1]) {
                let t = a[j]
                a[j] = a[j+1]
                a[j+1] = t
            }
        }
    }   
}
